From 88220e0ee41640940e7686fe0cab7f1e0bfb42f1 Mon Sep 17 00:00:00 2001
From: Jan Sikorski <jsikorski@codeweavers.com>
Date: Mon, 18 Jan 2021 14:18:17 +0100
Subject: [PATCH] winemac.drv: Include depth32+stencil pixel formats in
 enumeration.

This is especially important for M1 GPUs as they don't support the 24/8 format.

Signed-off-by: Jan Sikorski <jsikorski@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---
 dlls/winemac.drv/opengl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git dlls/winemac.drv/opengl.c dlls/winemac.drv/opengl.c
index b66af0f1c47..9f02e93d376 100644
--- dlls/winemac.drv/opengl.c
+++ dlls/winemac.drv/opengl.c
@@ -741,7 +741,8 @@ static void enum_renderer_pixel_formats(renderer_properties renderer, CFMutableA
 
                         if (!(renderer.stencil_modes & depth_stencil_modes[stencil_mode].mode))
                             continue;
-                        if (accelerated && depth_stencil_modes[depth_mode].bits != 24 && stencil_mode > 0)
+                        if (accelerated && depth_stencil_modes[depth_mode].bits != 24 &&
+                            depth_stencil_modes[depth_mode].bits != 32 && stencil_mode > 0)
                             continue;
 
                         attribs[n++] = kCGLPFAStencilSize;
